apmpackage/cmd/genpackage/main.go

```diff
@@ -0,0 +1,110 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/fs"
+	"log"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/elastic/beats/v7/libbeat/common"
+)
+
+var (
+	outputDir = flag.String("o", "", "directory into which the package will be rendered (required)")
+	version   = flag.String("version", "", "integration package version (required)")
+)
+
+func generatePackage(pkgfs fs.FS, version *common.Version) error {
+	// Walk files, performing some APM-specific validations and transformations as we go.
+	//
+	// We assume the target destination does not yet exist.
+	if err := fs.WalkDir(pkgfs, ".", func(path string, d fs.DirEntry, err error) error {
```

Comment:
We can return here directly, no need to handle the error, right?

Reply:
Indeed. I'll do that in my followup PR, in the interest of time.